pkg/src/geolocation/GeolocationBase.ts

```diff
@@ -0,0 +1,78 @@
+import axios from 'axios'
+
+/**
+ * Geo location abstraction
+ * having an address an endpoint and an this.apiKey,
+ * this class is able to calculate latitude and longitude coordinates with a 3rd party service
+ */
+export default abstract class GeolocationBase {
```

Comment:
Do you feel this class could have been a bit more decoupled and extensible?

Reply:
Sure, currently it is limited to GET requests only. I even placed an 'isPublic' property, if true we would not have the need for an api key. So yes, it could have more functionalities and thay can be added as the project requests to.

Having said that, I don't think that it could have been more extensible and decoupled, because an abstract class already is a base for behabiours to grow, so whoever extends it is free to grow as it feels, of course that all classes would be bound by a single core functionality, but that doesn't make them "coupled" in my eyes